src/application-insights/azext_applicationinsights/custom.py

```diff
@@ -75,6 +75,32 @@ def create_or_update_component(cmd, client, application, resource_group_name, lo
         raise ex
 
 
+def _is_workspace_centric(workspace):
+    return workspace.properties['features']['enableLogAccessUsingOnlyResourcePermissions'] is False
+
+
+def _is_resource_centric(workspace):
+    return workspace.properties['features']['enableLogAccessUsingOnlyResourcePermissions'] is True
+
+
+# Here are two cases that need users' consent during APM (Application Performance Management) migration:
+# 1. Bind a workspace-centric workspace to a classic AI (no workspace integration).
+# 2. Migrate a resource-centric workspace to a workspace-centric workspace for an AI.
+def _apm_migration_consent(cmd, new_workspace_resource_id, existing_workspace_resource_id):
+    from azure.cli.command_modules.resource.custom import show_resource
+    new_workspace = show_resource(cmd, [new_workspace_resource_id])
+    if _is_workspace_centric(new_workspace):
+        if existing_workspace_resource_id:
+            existing_workspace = show_resource(cmd, [existing_workspace_resource_id])
+            need_consent = _is_resource_centric(existing_workspace)
+        else:  # This is a classic AI which isn't binding to a log analytics workspace.
+            need_consent = True
+
+        if need_consent:
+            from azure.cli.core.util import user_confirmation
```

Comment:
Does user have to user terminal to execute this command?

Reply:
yes